Modules/Product/Http/Services/ProductService.php

```diff
@@ -788,21 +902,20 @@ public function update($request): array
             ->update([
                 'feature_key' => 1
             ]);
-        $productImage = ProductImage::where('product_id', $productId)->where('image_sort', $featuredKey)->first();
+        $productImage = Product::find($productId)->productImages()->where('image_sort', $featuredKey)->first();
         $featuredImage = $productImage->images;
         Product::where('id', $productId)
             ->update([
                 'featured_image' => $featuredImage
             ]);
-        $productFeatureImage = ProductImage::where('product_id', $productId)->where('feature_key', '1')->first();
+        $productFeatureImage = Product::find($productId)->productImages()->where('feature_key', '1')->first();
```

Comment:
use eager loading please. get the productImages in product query and then use from there.